compatibility_server/configs.py

```diff
@@ -12,7 +12,41 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-"""Common configs for compatibility_lib."""
+"""Common configs for compatibility_lib and compatibility_server.
+
+Note that a unit test exists for checking that the configs.py file in
+compatibility_lib is the same as the configs.py file in compatibility_server.
+
+The reason for this set up is that these modules need to be isolated from
+each other, but there also needs to be consistency in the objects and data
+in this file since they exist in the same workflow.
+"""
+
+
+import enum
+
+
+class PackageStatus(enum.Enum):
```

Comment:
I still don't understand why this is here?

Reply:
This will replace `PipCheckResultType` in compatibility_server and `Status` in compatibility_store. They are equivalent to each other in the current workflow and need to be expanded upon to support the changes listed in the PRD. I chose to create `PackageStatus` in this file since this file specifically contains items that need to be consistent among compatibility_store and compatibility_lib.

Reply:
I think that the backend should be, to the extent reasonable, independent of the frontend.

Reply:
As I understand it, the backend runs a pip check on a given package(s) and that is where the status is generated. The front end uses that status to serve a badge. I don't see how the backend status can be different from the front end status. In other words, because the check is done by the backend, if the possible statuses generated by the backend are [UNKNOWN, SUCCESS], the front end won't be able to serve a badge status of INCOMPATIBLE.

Before, compatibility_server and compatibility_lib independently defined enums to capture the possible statuses, but this is not good practice since if one thing changes in the backend, the frontend would break without the same change being applied.

Reply:
The backend has only 3 statuses: success, install_error, check_warning. It doesn't figure out if the dependencies are outdated, it doesn't care if it is checking one or more packages. And it doesn't know anything about storage (i.e. MISSING_DATA).

Reply:
But we're getting rid of install_error and check_warning and adding INTERNAL_ERROR, SELF_INCOMPATIBLE, and INCOMPATIBLE?

Reply:
But why are does it make sense to make this change? Most of this enum doesn't apply to the compatibility server but the result that it currently returns makes sense based on what it actually does.

Reply:
I thought of a compromise - take a look at the latest commit and let me know what you think.